### PR TITLE
Add release notes for PR #56243

### LINKED ISSUE #58907: "Cargo test" uses a thread pool in 1.33.0
Previous versions of Cargo would create and destroy a new thread for each test.  Cargo 1.33.0 instead creates a thread pool and reuses the same thread for multiple tests.  This can cause test failures for crates that implicitly rely on Cargo's old behavior for isolation.  I'm not saying that the new behavior is wrong, but it deserves a mention in the release notes.

https://github.com/nix-rust/nix/pull/1033

### PR DESCRIPTION
**Pull request by asomers**
Fixes #58907

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -80,6 +80,8 @@ Libraries
   `swap_bytes`, `from_be`, `from_le`, `to_be`, `to_le` are now `const` for all
   numeric types.][57234]
 - [`Ipv4Addr::new` is now a `const` function][57234]
+- [Libtest no longer creates a new thread for each test when
+  `--test-threads=1`.  It also runs the tests in deterministic order][56243]
```
Comment (Mark-Simulacrum) [inline RELEASES.md]: Could you move this to the compatibility notes section?

I don't think it's really a "feature"

**Comment by pietroalbini** [comment]
Please avoid the `[skip ci]`, we actually need CI to run to land PRs `:)`

**Comment by Mark-Simulacrum** [comment]
Thanks!

@bors r+ rollup

**Pull request merged**